core/src/main/java/org/mule/runtime/core/routing/requestreply/SimpleAsyncRequestReplyRequester.java

```diff
@@ -31,7 +31,8 @@
   protected void sendAsyncRequest(Event event) throws MuleException {
     event = Event.builder(event)
         .message(InternalMessage.builder(event.getMessage()).addOutboundProperty(MULE_REPLY_TO_PROPERTY, getReplyTo())
-            .addOutboundProperty(MULE_REPLY_TO_REQUESTOR_PROPERTY, flowConstruct.getName()).build())
+            .addOutboundProperty(MULE_REPLY_TO_REQUESTOR_PROPERTY, getLocation().getParts().get(0).getPartPath())
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.